agent/agent-bootstrap/src/main/java/com/microsoft/applicationinsights/agent/bootstrap/MainEntryPoint.java

```diff
@@ -102,6 +103,29 @@ public static void start(Instrumentation instrumentation, URL bootstrapURL) {
         }
     }
 
+    private static FriendlyException getFriendlyException(Throwable t) {
+        if (t instanceof FriendlyException) {
+            return (FriendlyException) t;
+        }
+        Throwable cause = t.getCause();
+        if (cause == null) {
+            return null;
+        }
+        return getFriendlyException(cause);
```

Comment:
what is the point of this recursion here?

Reply:
In the MainEntryPoint class there are different kind of exceptions thrown and FriendlyException is chained as part of these exceptions. We have to look for Exception where the cause if FriendlyException.

Reply:
my concern is will this lead to an infinite loop when throws is not FriendlyException?

Reply:
If there is no FriendlyException, the cause will eventually be null and the function returns null. Will try to add a unit test for this case.

Reply:
added unit test for getFriendlyException method